src/Lti/LoadBalancer/AbstractLtiInstanceLoadBalancer.php

```diff
@@ -32,7 +33,7 @@ public function __construct(array $ltiInstances)
         $this->ltiInstances = $ltiInstances;
     }
 
-    public function getLoadBalancedLtiInstanceUrl(string $value): string
+    protected function getLoadBalancedLtiInstanceUrl(string $value): string
```

Comment:
I'd go with keeping it a separated class 
it will increase flexibility in case we need to implement different algorithms 
IMHO algorithms should be unittested

Reply:
Yes you are right, but the fact we have an injected load balancer here is a hack. We've a ticket somewhere in the backlog to extract the load balancer from here to a separate application, so I would not spend time now on it (YAGNI) if you don't mind.